### PR TITLE
[ci] refactor production deployment

### PR DESCRIPTION
**Pull request by alvicsam**
- removed tags refs
- deploy to `parity-prod` instead `parity-chains`
- removed ci part from dockerfile
- removed old helm-chart

cc https://github.com/paritytech/ci_cd/issues/544
close https://github.com/paritytech/ci_cd/issues/690

### CODE REVIEW

File: helm/values-parity-prod.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+gcloud-sqlproxy:
```
[the remark this answers is not in the document]
Comment (mutantcornholio) [reply helm/values-parity-prod.yaml:1]: All the environment variables, like `DB_PORT`, `DB_PASSWORD` and so on aren't used anywhere in code

[the remark this answers is not in the document]
Comment (alvicsam) [reply helm/values-parity-prod.yaml:1]: https://github.com/paritytech/ci_cd/issues/690

[the remark this answers is not in the document]
Comment (mutantcornholio) [reply helm/values-parity-prod.yaml:1]: ah, okay, sounds good

**Review by mutantcornholio — APPROVED** [review]
❤️

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -31,57 +30,23 @@ default:
...
-      buildah login --username "$Docker_Hub_User_Parity" --password-stdin docker.io
-    - buildah bud
-      --format=docker
-      --target=$TARGET
-      --tag "$IMAGE_NAME:$DOCKER_TAG"
-      --file $DOCKERFILE .
-    - buildah push --format=v2s2 "$IMAGE_NAME:$DOCKER_TAG"
-  after_script:
-    - buildah logout --all
+      when: manual
```
Comment (mordamax) [inline .gitlab-ci.yml:49]: i don't fully get why the prod refs are `master` + manual? could you please tech me how to read it ?

File: helm/values.yaml [hunk 3]
```diff
@@ -1,96 +1,61 @@
...
+    annotations: {}
+    storageClass: hdd-csi
+    size: 100G
+    accessModes:
+      - ReadWriteOnce
+  env:
+    GITLAB_PUSH_NAMESPACE: parity/mirrors
+    GITLAB_JOB_IMAGE: paritytech/ci-linux:production
+    GITLAB_DOMAIN: gitlab.parity.io
+    PIPELINE_SCRIPTS_REPOSITORY: https://github.com/paritytech/pipeline-scripts
```
Comment (mordamax) [inline helm/values.yaml:46]: https://github.com/paritytech/command-bot-scripts/

Comment (mordamax) [inline helm/values.yaml:46]:
```suggestion
    PIPELINE_SCRIPTS_REPOSITORY: https://github.com/paritytech/command-bot-scripts/
```

File: .gitlab-ci.yml (hunk 2 above)
Comment (alvicsam) [reply .gitlab-ci.yml:49]: This means that you can deploy to prod on master branch but you have to press the "deploy" button

File: helm/values.yaml (hunk 3 above)
Comment (alvicsam) [reply helm/values.yaml]: added in https://github.com/paritytech/command-bot/pull/149/commits/a0e119a01d8b1dc9888b86855cf38502cd2c08cb

File: .gitlab-ci.yml (hunk 2 above)
Comment (mordamax) [reply .gitlab-ci.yml:49]: but that's not what we agreed to :) 

prod should be deployable by tagging new version `/^v[0-9]+\.[0-9]+.*$/             # i.e. v1.0, v2.1rc1` 
staging  either `/^stg-v[0-9]+\.[0-9]+.*$/` or master

now it looks like it's prod manually through gitlab and staging automatically from master, but sometimes I need to check some dev branch on stage, so only master doens't work

Comment (mordamax) [reply .gitlab-ci.yml:52]: yes, i think what i just wrote above corresponds to your suggestion

- ✅  automatic deployments from master to staging on merge (+ optionally manual ones, so we can test dev-branches for some apps)
- ✅ production deployment requiring a manual trigger (tagging or using github releases is a manual trigger. it also generates automatically the changelog)

Comment (mordamax) [reply .gitlab-ci.yml:52]: i thought when you mention "creating special tags" you mean how it was in the past, that staging was allowed only when you tag, but not from master. So many projects recently had a "master" ref for staging as well, which were auto-deploying the **staging**

for **prod** - we could also deploy it from button, but that wouldn't give a sense of what was releasing and when, so I think using GH releases with history of changelog is better option. At least, if not GH releases, then tagging would give a minimum sense of "when" and "what" you can infer from comparing diffs from tags.
So when you were saying "requiring a manual trigger" I didn't think about a button necessary, but GH releases, which is few clicks, pretty convenient and controlled

**Review by oleg-plakida — APPROVED** [review]

**Review by sergejparity — APPROVED** [review]

**Review by mordamax — APPROVED** [review]

**Pull request merged**